src/clj/muuntaja/middleware.clj

```diff
@@ -37,19 +37,18 @@
 (defn wrap-params [handler]
   "Middleware that merges request `:body-params` into `:params`.
   Supports async-ring."
-  (fn
-    ([request]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request)))
-    ([request respond raise]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request respond raise)))))
+  (letfn [(set-params
+            ([request]
+             (cond
+               (not (map? (:body-params request))) request
```

Comment:
Hash lookups are also slow so this version is now actually slower than the previous. could you assign both params in a let before the cond? despite the `params` is only used once, preassigning it too makes the code run much faster.